### PR TITLE
[Feature][Task-13] - BE: add basic docker image

### LINKED ISSUE #13: [BE] Add basic docker image
- Add docker folder on the root level(src)
- Add dockerfile with python alpine basic image
- Dockerfile must contain:
     - apt update and apt upgrade
     - copy files locally to docker image, files from src
     - copy entrypoint.sh, add writes permission
 - Verify docker run + ports works and you can get version endpoint

### PR DESCRIPTION
**Pull request by vasylboliuk**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: '3.8'
+
+services:
+  app:
+    build:
+      context: .
+      dockerfile: docker/Dockerfile
+    ports:
+      - "8000:8000"
```
Comment (alexgrand) [inline docker-compose.yml]: @vasylboliuk could you please add an endline

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+FROM python:3.12-alpine
+
+# Set the working directory
+WORKDIR /app
+
+# Set environment variables
+ENV PYTHONUNBUFFERED=1 \
+    POETRY_VERSION=1.8.3 \
+    POETRY_HOME="/opt/poetry" \
+    PATH=$PATH:/opt/poetry/bin
```
Comment (alexgrand) [inline docker/Dockerfile]: @vasylboliuk, it is more often used to divide env variable definitions for several layers, so if one env is changed, others don't require rehashing. Please divide these variables into several ENV and for PATH variable reuse the POETRY_HOME variable

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Use the Python Alpine base image
+FROM python:3.12-alpine
+
+# Set the working directory
+WORKDIR /app
```
Comment (alexgrand) [inline docker/Dockerfile:5]: @vasylboliuk please add /app as an ENV variable, not only as a workdir. This way you will be able to reuse it in your Dockerfile. Please change places where you use /app with env variable

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+# Copy Poetry project files (pyproject.toml and poetry.lock) first for caching dependencies
+COPY pyproject.toml poetry.lock /app/
+
+# Install dependencies using Poetry (without dev dependencies)
+
+RUN poetry --version
+RUN poetry lock --no-update
+RUN poetry install --no-root --only main
+RUN poetry show uvicorn
```
Comment (alexgrand) [inline docker/Dockerfile]: @vasylboliuk you could use just a `poetry check`

File: docker/entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+# Ruun some setup tasks before starting the main application
+echo "Setting up the environment..."
+
+# Print the directory structure for debugging purposes
+echo "File structure:"
+find /app
```
Comment (alexgrand) [inline docker/entrypoint.sh]: @vasylboliuk you could use `tree` which has way more functionality and describes project structure better

File: docker/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+# Run the main application
+echo "Starting the application..."
+exec poetry run uvicorn src.main:app --host 0.0.0.0 --port 8000 --reload
```
Comment (alexgrand) [inline docker/entrypoint.sh]: @vasylboliuk isn't it working with the simple sh/bash instead of exec?

**Review by alexgrand — CHANGES_REQUESTED** [review]
I like how it looks like. I have several comments and questions, but nothing major.

File: docker-compose.yml (hunk 1 above)
Comment (vasylboliuk) [reply docker-compose.yml]: Done

File: docker/Dockerfile (hunk 2 above)
Comment (vasylboliuk) [reply docker/Dockerfile]: Done

**Pull request merged**